src/libraries/Microsoft.PowerFx.Core/Types/DTypeExtensionsCore.cs

```diff
@@ -81,5 +81,30 @@ public static bool ContainsDataEntityType(this DType self, DPath path, int curre
             return self.GetNames(path).Any(n => n.Type.IsExpandEntity ||
                 (n.Type.IsAggregate && n.Type.ContainsDataEntityType(DPath.Root, currentDepth - 1)));
         }
+
+        /// <summary>
+        /// Try to get the column logical name.
+        /// </summary>
+        /// <param name="self"></param>
+        /// <param name="displayName"></param>
+        /// <param name="logicalName"></param>
+        /// <returns></returns>
+        public static bool TryGetLogicalName(this DType self, DName displayName, out DName logicalName)
```

Comment:
>TryGet [](http://example.com/codeflow?start=27&length=6)

In TryGetXXX methods, if it returns true, then the out parameter is set to null (or the default for that type). We shouldn't diverge from the "common knowledge" except in rare cases (otherwise it may make it confusing when reading the code).